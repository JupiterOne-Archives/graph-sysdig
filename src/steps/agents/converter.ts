import {
  createIntegrationEntity,
  Entity,
} from '@jupiterone/integration-sdk-core';

import { SysdigAgent } from '../../types';
import { Entities } from '../constants';

export function createAgentEntity(data: SysdigAgent): Entity {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _key: `sysdig-agent:${data.clusterName}-${data.machineId}-${data.labels.provider}-${data.labels.hostname}`,
        _type: Entities.AGENT._type,
        _class: Entities.AGENT._class,
        name: data.labels.hostname,
        agentStatus: data.agentStatus,
        clusterName: data.clusterName,
        accountId: data.labels.accountId,
        hostname: data.labels.hostname,
        machineId: data.labels.machineId,
        provider: data.labels.provider,
        agentConnectString: data.moreOptions.agentConnectString,
        category: ['application'],
      },
    },
  });
}
