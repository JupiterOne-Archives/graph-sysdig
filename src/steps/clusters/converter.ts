import {
  createIntegrationEntity,
  Entity,
  parseTimePropertyValue,
} from '@jupiterone/integration-sdk-core';

import { SysdigCluster } from '../../types';
import { Entities } from '../constants';

export function getClusterKey(provider: string, name: string) {
  return `sysdig-cluster:${provider}-${name}`;
}

export function createClusterEntity(data: SysdigCluster): Entity {
  const { agentConnectString, ...cluster } = data;

  return createIntegrationEntity({
    entityData: {
      source: {
        ...cluster,
      },
      assign: {
        _key: getClusterKey(data.provider, data.name),
        _type: Entities.CLUSTER._type,
        _class: Entities.CLUSTER._class,
        customerId: data.customerID,
        accountId: data.accountID,
        provider: data.provider,
        name: data.name,
        region: data.region,
        zone: data.zone,
        agentConnected: data.agentConnected,
        createdOn: parseTimePropertyValue(data.createdAt),
        nodeCount: data.nodeCount,
        clusterResourceGroup: data.clusterResourceGroup,
        version: data.version,
      },
    },
  });
}
