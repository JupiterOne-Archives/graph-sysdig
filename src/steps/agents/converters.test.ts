import { getMockAgent } from '../../../test/mocks';
import { createAgentEntity } from './converter';

describe('#createAgentEntity', () => {
  test('should convert to entity', () => {
    expect(createAgentEntity(getMockAgent())).toMatchSnapshot();
  });
});
