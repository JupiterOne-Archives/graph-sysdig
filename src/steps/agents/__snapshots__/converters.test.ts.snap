// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createAgentEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Scanner",
  ],
  "_key": "sysdig-agent:gcp-gke-sysdig-cluster-default-pool-12345678-6l1s",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "agentIssues": Array [
          Object {
            "explanation": "Agent was never connected.",
            "fix": "review agent logs and/or network connectivity",
            "reason": "Never Connected",
            "status": "Never Connected",
          },
        ],
        "agentStatus": "Never Connected",
        "clusterName": "sysdig-cluster",
        "labels": Object {
          "accountId": "1234567890",
          "hostname": "gke-sysdig-cluster-default-pool-12345678-6l1s",
          "machineId": "",
          "provider": "gcp",
        },
        "moreOptions": Object {
          "agentConnectString": "agent connect string",
        },
      },
    },
  ],
  "_type": "sysdig_agent",
  "accountId": "1234567890",
  "agentConnectString": "agent connect string",
  "agentStatus": "Never Connected",
  "category": Array [
    "application",
  ],
  "clusterName": "sysdig-cluster",
  "createdOn": undefined,
  "displayName": "gke-sysdig-cluster-default-pool-12345678-6l1s",
  "hostname": "gke-sysdig-cluster-default-pool-12345678-6l1s",
  "machineId": "",
  "name": "gke-sysdig-cluster-default-pool-12345678-6l1s",
  "provider": "gcp",
}
`;
